### PR TITLE
Catch up with amino 0.13.0

### PR DESCRIPTION
**Pull request by Liamsi**
Depends on #2687 
related: https://github.com/tendermint/go-amino/pull/222
related: https://github.com/tendermint/tendermint/pull/2687
related: https://github.com/tendermint/go-amino/pull/233

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: blockchain/store.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (bs *BlockStore) LoadBlock(height int64) *types.Block {
 		part := bs.LoadBlockPart(height, i)
 		buf = append(buf, part.Bytes...)
 	}
-	err := cdc.UnmarshalBinary(buf, block)
+	err := cdc.UnmarshalBinaryLengthPrefixedBinary(buf, block)
```
Comment (Liamsi) [inline blockchain/store.go]: Oh no, this should actually be named: `UnmarshalBinaryLengthPrefixed` instead 🙈 Not sure how me and two reviewers didn't see that. I'll delete the release 0.13.0 for now and add commit to master that fixes that (and tag this again).

**Comment by ebuchman** [comment]
Should we merge  #2687 first?

**Comment by Liamsi** [comment]
Yes, we should. Otherwise we'd probably have to redo the changes from #2687 here, too.

File: blockchain/store.go (hunk 1 above)
Comment (Liamsi) [reply blockchain/store.go]: https://github.com/tendermint/go-amino/pull/236

**Pull request merged**